tests/core/test_flyer.py

```diff
@@ -0,0 +1,272 @@
+    return SameTriggerDetectorGroupLogic(controllers, writers)
+
+
+async def test_hardware_triggered_flyable(
+    RE: RunEngine, detector_group: SameTriggerDetectorGroupLogic
+):
+    names = []
+    docs = []
+    RE.subscribe(lambda name, _: names.append(name))
+    RE.subscribe(lambda _, doc: docs.append(doc))
+
+    trigger_logic = DummyTriggerLogic()
+    flyer = HardwareTriggeredFlyable(detector_group, trigger_logic, [], name="flyer")
+
+    def flying_plan():
+        yield from bps.stage_all(flyer)
+        assert trigger_logic.state == TriggerState.stopping
+
+        yield from bps.open_run()
+
+        yield from bps.kickoff(flyer)
+
+        yield from bps.complete(flyer, wait=False, group="complete")
+        assert trigger_logic.state == TriggerState.starting
+
+        done = False
+        while not done:
+            try:
+                yield from bps.wait(group="complete", timeout=0.5)
+            except TimeoutError:
+                pass
+            else:
+                done = True
+
+            yield from bps.collect(
+                flyer, stream=True, return_payload=False, name="primary"
+            )
+            yield from bps.sleep(0.001)
+        yield from bps.wait(group="complete")
+        yield from bps.close_run()
+
+        yield from bps.unstage_all(flyer)
+        for controller in detector_group.controllers:
+            assert controller.disarm.called  # type: ignore
+            assert controller.disarm.call_count == 3  # type: ignore
+        assert trigger_logic.state == TriggerState.stopping
+
+    # move the flyer to the correct place, before fly scanning.
+    RE(bps.mv(flyer, 1))
+    assert trigger_logic.state == TriggerState.preparing
+    for controller in detector_group.controllers:
+        assert controller.disarm.called  # type: ignore
+        assert controller.disarm.call_count == 1  # type: ignore
+        assert controller.arm.called  # type: ignore
+        assert controller.arm.call_count == 1  # type: ignore
+
+    # fly scan
+    RE(flying_plan())
+
+    assert names == [
+        "start",
+        "descriptor",
+        "stream_resource",
+        "stream_datum",
+        "stream_resource",
+        "stream_datum",
+        "stop",
+    ]
+
+
+async def test_scan_spec_flyable_wont_pause_if_not_flying(
+    RE: RunEngine, detector_group: SameTriggerDetectorGroupLogic
+):
+    trigger_logic = DummyTriggerLogic()
+    flyer = ScanSpecFlyable(detector_group, trigger_logic, [], name="flyer")
+
+    with pytest.raises(AssertionError):
+        await flyer.pause()
+
+    with pytest.raises(AssertionError):
+        await flyer.resume()
+
+    # now check that you can do the first half of a plan, pause it, and it unstages and prepares again.
+
+
+""" For the moment, this test doesnt work.
+
+This is because you can't add a scanspec Path and an int.
+Flyers set a _current_frame to 0 by default, but then expect to be able to add
+any generic T value to it from _prepare...
+"""
```

Comment:
Feel free to exclude the scanspec bit from this PR, it's not required for the first i22 experiment, but I thought it was when I started...

Reply:
Well that'll help the test coverage, so will do for now :+1: